### PR TITLE
Adsk Contrib - Improve the library install step

### PR DESCRIPTION
**Pull request by hodoulp**
Signed-off-by: Patrick Hodoul <Patrick.Hodoul@autodesk.com>

The pull request improves the library installation step following comments from @elsandosgrande and @Begasus  in #1415. Even if the comments focus on the 2.0.x release, I moved the fixes in `main` to later move them to `RB-2.1` (i.e. 2.1.x is now the official release).

From the root directory, the ways to test are:

- `cmake -B ./build_rls -S . -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=./install_rls -DCMAKE_INSTALL_BINDIR=binDir -DCMAKE_INSTALL_DATADIR=dataDir -DCMAKE_INSTALL_LIBDIR=libDir -DCMAKE_INSTALL_INCLUDEDIR=includeDir -DBUILD_SHARED_LIBS=ON`
- `make -C ./build_rls install -j 12`
- `ls -all ./install_rls`

and

- `cmake -B ./build_rls -S . -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=./install_rls -DBUILD_SHARED_LIBS=ON`
- `make -C ./build_rls install -j 12`
- `ls -all ./install_rls`

Tests were done on MacOS 10.15.7 with cmake 3.20 and AppleClang 12.0.

### CODE REVIEW

**Comment by hodoulp** [comment]
The Windows CI build compiling the documentation fails at the setup step because of `doxygen`:
```
Progress: 100% - Completed download of C:\Users\runneradmin\AppData\Local\Temp\chocolatey\doxygen.install\1.9.3\doxygen-1.9.3-setup.exe (41.22 MB).
Download of doxygen-1.9.3-setup.exe (41.22 MB) completed.
Error - hashes do not match. Actual value was '83F7241DFB44C5BB271072792379BCB539284247'.
ERROR: Checksum for 'C:\Users\runneradmin\AppData\Local\Temp\chocolatey\doxygen.install\1.9.3\doxygen-1.9.3-setup.exe' did not meet '24fd29b4476ad1197a41438efcd67b926db923cd' for checksum type 'sha1'. Consider passing the actual checksums through with --checksum --checksum64 once you validate the checksums are appropriate. A less secure option is to pass --ignore-checksums if necessary.
The install of doxygen.install was NOT successful.
Error while running 'C:\ProgramData\chocolatey\lib\doxygen.install\tools\chocolateyInstall.ps1'.
 See log for details.

Chocolatey installed 0/1 packages. 1 packages failed.
 See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).

Failures
 - doxygen.install (exited -1) - Error while running 'C:\ProgramData\chocolatey\lib\doxygen.install\tools\chocolateyInstall.ps1'.
 See log for details.
Error: Process completed with exit code 127.
```

**Comment by Begasus** [comment]
In the process with other stuff atm, once done I'll check these changes on Haiku.

**Comment by remia** [comment]
This Chocolatey Doxygen package generate a lot of CI errors over time, it would be nice to find a replacement.

A question about the new variables for the ExternalProject built dependencies, are these fixing a known issue or is it just a general improvement? I can't see how it matters for these temporary built libraries?

**Comment by hodoulp** [comment]
> A question about the new variables for the ExternalProject built dependencies, are these fixing a known issue or is it just a general improvement? I can't see how it matters for these temporary built libraries?

The first case fails to compile without the changes because there is a mismatch between the expected `libDir` vs. `lib` sub-directories when linking with the external third-party libraries.

**Comment by hodoulp** [comment]
> This Chocolatey Doxygen package generate a lot of CI errors over time, it would be nice to find a replacement.

As long as there is at least one successful compilation including the documentation, the CI build is fine at my point of view. There are currently three documentation compilations on Linux, one on macOS and one on Windows. Can we disable docs for Windows & macOS (i.e. only keep the Linux one)?

**Comment by remia** [comment]
> > A question about the new variables for the ExternalProject built dependencies, are these fixing a known issue or is it just a general improvement? I can't see how it matters for these temporary built libraries?
> 
> The first case fails to compile without the changes because there is a mismatch between the expected `libDir` vs. `lib` sub-directories when linking with the external third-party libraries.

Good point, I forgot that we are using `CMAKE_INSTALL_*` in the imported target library location when building third party.

> Can we disable docs for Windows & macOS and only keep one on Linux?

Assuming RTD uses Linux build, I guess it should be fine.

**Review by remia — APPROVED** [review]

**Comment by remia** [comment]
Other workflows like analysis and even wheels build doc on Windows, and for the latter I think it's needed to have proper docstring in the python wheel. So there is still a need to find a fix at some point, here I believe the problem will solve by itself in a few days at most (I'll keep an eye on the nightly build to find out when).

**Comment by remia** [comment]
Looks like the issue was fixed as the nightly build succeeded.

File: .github/workflows/ci_workflow.yml [hunk 1]
```diff
@@ -471,7 +471,7 @@ jobs:
           - build: 5
             build-type: Release
             build-shared: 'ON'
-            build-docs: 'ON'
+            build-docs: 'OFF'
```
Comment (doug-walker) [inline .github/workflows/ci_workflow.yml:474]: Is this necessary?  (It means we would not have any tests on Windows that build docs.)

Comment (hodoulp) [reply .github/workflows/ci_workflow.yml:474]: In GitHub Action, there are frequent Chocolatey `Doxygen` download failures on Windows platform only. It then creates a build failure blocking any merge. So, the change disables the docs build for Windows and only preserves one docs build on Linux.

**Review by doug-walker — APPROVED** [review]

**Pull request merged**